arraycontext/container/traversal.py

```diff
@@ -987,4 +989,79 @@ def treat_as_scalar(x: Any) -> bool:
 
 # }}}
 
+
+# {{{
+
+def bcast_left(
+            op: Callable[[ArrayOrArithContainer, ArrayOrArithContainer],
+                    ArrayOrArithContainer],
+            left: ArrayOrArithContainer,
+            right: ArithArrayContainer,
+        ) -> ArrayOrArithContainer:
+    try:
+        serialized = serialize_container(right)
+    except NotAnArrayContainerError:
+        return op(left, right)
+
+    return deserialize_container(right, [
+        (k, op(left, right_v)) for k, right_v in serialized])
+
+
+def bcast_right(
+            op: Callable[[ArrayOrArithContainer, ArrayOrArithContainer],
+                    ArrayOrArithContainer],
+            left: ArrayOrArithContainer,
+            right: ArithArrayContainer,
+        ) -> ArrayOrArithContainer:
+    try:
+        serialized = serialize_container(left)
+    except NotAnArrayContainerError:
+        return op(left, right)
+
+    return deserialize_container(right, [
+        (k, op(left_v, right)) for k, left_v in serialized])
+
+
+def bcast_left_until_actx_array(
```

Comment:
Argument order? (Here and above.) I initially had the operator in the middle but went with "first" to allow convenient use of `partial`.

Reply:
I like the functional feel of it too :+1: